opencog/openpsi/utilities.scm

```diff
@@ -326,13 +326,14 @@
 		result
 		; else check if entity is an evaluation or predicate or schema
 		(let ((type (cog-type entity)))
-			(if (or (equal? type 'PredicateNode)
-				   (equal? type 'GroundedPredicateNode)
+			(if (or (equal? type 'GroundedPredicateNode)
 				   (equal? type 'DefinedPredicateNode))
 				(set! result (cog-evaluate! (Evaluation entity (List)))))
-			(if (or (equal? type 'SchemaNode)
+			(if (equal? type 'PredicateNode)
+				(set! result (cog-tv entity)))
+			(if (or ; (equal? type 'SchemaNode) ;FIXME SchemaNode not executable
```

Comment:
@AmeBel, you are right, a SchemaNode is not valid here. Thanks for the catch. 
I am not using SchemaNode in the psi dynamics code, but I was trying to think ahead here and handle cases where the dynamics code is needing to get values for entities defined in other modules, so trying to cover the bases. But yes, you are right that SchemaNode should not be one of those, thanks!